ironfish/src/wallet/wallet.ts

```diff
@@ -294,7 +294,9 @@ export class Accounts {
     await this.db.database.transaction(async (tx) => {
       await this.resetAccounts(tx)
       this.chainProcessor.hash = null
-      await this.saveAccountsToDb(tx)
+      await this.db.clearDecryptedNotes(tx)
+      await this.db.clearNullifierToNoteHash(tx)
+      await this.db.clearTransactions(tx)
```

Comment:
I think these 3 lines should probably go into resetAccounts. My intuition is that I ask myself, "What is resetAccounts responsibility? I don't think reset accounts should ever be called without these 3 lines accompanying it, so that's why I think they should be together.

Reply:
I think the only use case for resetting the in-memory account storage without clearing the database is in tests

I'll update